plugins/examples/security-authorization-engine/build.gradle

```diff
@@ -11,19 +12,24 @@ esplugin {
   noticeFile rootProject.file('NOTICE.txt')
 }
 
+// let the javaRestTest see the classpath of main
+GradleUtils.extendSourceSet(project, "main", "javaRestTest")
```

Comment:
Why is this necessary here but not done by default in the java rest test plugin?

Since this is an example project we should use idiomatic patterns as much as possible. If this is required because of some kind of nasty test implementation we should refactor this.

@rjernst thoughts? As I understand it a Java REST test should be no different than a YAML one in that the only interactions should be via the REST API, therefore there should be no need for plugin code itself to be on the test classpath.

Reply:
The examples are not exactly idomatic to start with, they could use some updating. In this case the test really does need to see the main classes, which should probably be converted to a `ESSingleNodeTestCase` or the tests written as pure black box ?  I can log an issue, but probably outside the scope of this change.

Reply:
Yeah, I don't like the idea of `GradleUtils` being used in a plugin example. That's not intended to be "public". Worst case if the tests are structure such that they strictly need those classes, we should just extend the source set in the build script explicilty rather than leverage this utility method.

Reply:
For this case I am able to simply add the dependency to the main runtimeClasspath. https://github.com/elastic/elasticsearch/pull/59444/commits/ef3466837535242c2d0c4be28241bde6ff2af951
```
javaRestTestImplementation project.sourceSets.main.runtimeClasspath
```